### PR TITLE
 feat: [AUTH-4354] Add B2B Session JWT Methods

### PR DESCRIPTION
**Pull request by max-stytch**
This PR adds the B2B equivalent of #11 along with RBAC Authorization caching + helpers.

This PR requires us to add some more secrets to CI - will ask for Infra help with that on Monday.

### CODE REVIEW

File: Stytch.net/Clients/consumer/M2M.cs [hunk 1]
```diff
@@ -14,15 +14,14 @@
 using System.Threading.Tasks;
 
 
-
-
```
Comment (max-stytch) [inline Stytch.net/Clients/consumer/M2M.cs:18]: TODO: Determine what is up with these whitespace changes. I think we need to add an [editorconfig](https://learn.microsoft.com/en-us/visualstudio/ide/create-portable-custom-editor-options?view=vs-2022) file - which I'll do in a future PR

File: Stytch.net/Utility/Utility.cs [hunk 2]
```diff
@@ -6,98 +6,139 @@
 using System.Threading.Tasks;
 using Microsoft.IdentityModel.Tokens;
 using Stytch.net.Clients;
+using Stytch.net.Models.Consumer;
 
-public static class Utility
```
Comment (max-stytch) [inline Stytch.net/Utility/Utility.cs:10]: Sorry for the diff - everything got indented here

File: Stytch.net/Utility/Utility.cs [hunk 3]
```diff
@@ -6,98 +6,139 @@
...
+            return new AuthenticateJwtResult()
+            {
+                Subject = (string)result.Claims[ClaimTypes.NameIdentifier],
+                CustomClaims = customClaims
+            };
+        }
+
+        private static async Task<JsonWebKeySet> GetJwksFromUrl(HttpClient client, string jwksUrl)
+        {
+            if (_cachedJwks == null || _cachedJwksUrl != jwksUrl || DateTime.UtcNow - _jwksLastFetched > _jwksRefreshInterval)
```
Comment (max-stytch) [inline Stytch.net/Utility/Utility.cs:103]: Small change - we now invalidate if the jwksUrl doesn't match what we saw previously. This mostly helps with supporting tests that access both B2B and B2C JWKs

File: Stytch.net/Utility/Utility.cs [hunk 4]
```diff
@@ -6,98 +6,139 @@
...
 
-    private static async Task<JsonWebKeySet> GetJwksFromUrl(HttpClient client, string jwksUrl)
-    {
-        if (_cachedJwks == null || DateTime.UtcNow - _jwksLastFetched > _jwksRefreshInterval)
+        private static async Task<RbacPolicy> GetRbacPolicy(IPolicyGetter policyGetter)
         {
-            string jwksJson = await client.GetStringAsync(jwksUrl);
-            _cachedJwks = new JsonWebKeySet(jwksJson);
-            _jwksLastFetched = DateTime.UtcNow;
+            if (_cachedRbacPolicy == null || DateTime.UtcNow - _rbacPolicyLastFetched > _rbacRefreshInterval)
```
Comment (max-stytch) [inline Stytch.net/Utility/Utility.cs:132]: RBAC policy cache logic is nearly identical to JWK cache logic.

File: Stytch.net.Tests/Clients/B2BSessions.cs [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+            AuthorizationCheck = new AuthorizationCheck
+            {
+                OrganizationId = null,
+                ResourceId = null,
+                Action = null
+            }
+        };
+
+        // Act
+        await b2bClient.Sessions.AuthenticateJwtLocal(request);
```
Comment (ollie-stytch) [inline Stytch.net.Tests/Clients/B2BSessions.cs:93]: Nothing to assert here?

File: Stytch.net.Tests/Utility/RbacPolicy.cs [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+                        Actions = new List<string>() { "read" },
+                        ResourceId = "images",
+                    }
+                }
+            }
+        }
+    };
+
+    [Theory]
+    [InlineData(new string[] { "default", "reader" }, "documents", "read", true)]
```
Comment (ollie-stytch) [inline Stytch.net.Tests/Utility/RbacPolicy.cs:102]: 👀

File: Stytch.net/Clients/b2b/Sessions.cs [hunk 7]
```diff
@@ -40,10 +41,11 @@ B2BSessionsGetRequest request
...
             };
-            uriBuilder.Query = Utility.BuildQueryString(new Dictionary<string, string> {
-            {"organization_id", request.OrganizationId},
-            {"member_id", request.MemberId},
-        });
+            uriBuilder.Query = Utility.BuildQueryString(new Dictionary<string, string>
+            {
+                { "organization_id", request.OrganizationId },
+                { "member_id", request.MemberId },
+            });
```
Comment (ollie-stytch) [inline Stytch.net/Clients/b2b/Sessions.cs:48]: If this is also gonna have the whitespace issue maybe we can do the same fix we had in the previous PR?

File: Stytch.net/Models/B2B/Sessions.cs [hunk 8]
```diff
@@ -633,4 +634,98 @@ public enum B2BSessionsExchangeRequestLocale
         [EnumMember(Value = "pt-br")]
         PTBR,
     }
+
+    // MANUAL(AuthenticateJWT)(TYPES)
```
Comment (ollie-stytch) [inline Stytch.net/Models/B2B/Sessions.cs:638]: `B2BAuthenticateJWT`? Not sure if it matters these being identical across consumer and b2b

File: Stytch.net.Tests/Clients/B2BSessions.cs (hunk 5 above)
Comment (max-stytch) [reply Stytch.net.Tests/Clients/B2BSessions.cs:93]: We're asserting that an error isn't thrown! (and the test is skipped for now anyway...)

File: Stytch.net/Clients/b2b/Sessions.cs (hunk 7 above)
Comment (max-stytch) [reply Stytch.net/Clients/b2b/Sessions.cs:48]: Hopefully the editorconfig fixes it. We'll see

**Pull request merged**